.github/workflows/pr.yml

```diff
@@ -36,4 +39,17 @@ jobs:
       # we run our artifact directly please use the prebuild
       # elastic/docs-builder@main GitHub Action for all other repositories!
       - name: Build documentation
-        run: .artifacts/publish/docs-builder/release/docs-builder --strict
+        env:
+          PR_NUMBER: ${{ github.event.pull_request.number }}
+        run: .artifacts/publish/docs-builder/release/docs-builder --strict --path-prefix "/${GITHUB_REPOSITORY}/pull/${PR_NUMBER}"
+      
+      - uses: actions/upload-artifact@v4
+        with:
+          name: docs
+          path: .artifacts/docs/html
```

Comment:
Cache our binary instead of the html. 

Our html is small but for e.g docs-content it will be many megabytes (e.g just the output of the converted obs docs is 100mb zipped due to all the images). 

Secondly we can then either grab the binary or run our github action inside the preview action. 

With the preview github action being the only one we expose to everyone to add to their PR builds.